jujugui/static/gui/src/app/components/profile/model-list/model-list.js

```diff
@@ -0,0 +1,256 @@
+    };
+  }
+
+  componentWillMount() {
+    this._fetchModels(this.props.facadesExist);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const props = this.props;
+    if (props.userInfo.profile !== nextProps.userInfo.profile ||
+        props.facadesExist !== nextProps.facadesExist) {
+      this._fetchModels(nextProps.facadesExist);
+    }
+  }
+
+  /**
+    Makes a request of the controller to fetch the user's availble models.
+    @param {Boolean} facadesExist Whether the controller is connected or not.
+  */
+  _fetchModels(facadesExist) {
+    if (!facadesExist) {
+      console.warn('Controller not connected, skipping fetching models.');
+      return;
+    }
+    // Delay the call until after the state change to prevent race
+    // conditions.
+    this.setState({loadingModels: true}, () => {
+      this.props.listModelsWithInfo(this._fetchModelsCallback.bind(this));
+    });
+  }
+
+  /**
+    Callback for the controller listModels call.
+    @param {String} err The error from the request, or null.
+    @param {Array} modelList The list of models.
+  */
+  _fetchModelsCallback(err, modelList) {
+    this.setState({loadingModels: false}, () => {
+      if (err) {
+        const message = 'Cannot load models';
+        console.error(message, err);
+        this.props.addNotification({
+          title: message,
+          message: `${message}: ${err}`,
+          level: 'error'
+        });
+        return;
+      }
+      // Split the models into ones owned by the active user and ones
+      // shared to the active user.
+      const models = {
+        owned: [],
+        shared: []
+      };
+      modelList.forEach(model => {
+        if (model.owner === `${this.props.userInfo.profile}@external`) {
+          models.owned.push(model);
+        } else {
+          models.shared.push(model);
+        }
+      });
+      this.setState({models});
+    });
+  }
+
+  /**
+    Generates the list of models that are owned by the active user.
+    @return {Object} The model list as JSX.
+  */
+  _generateMyModels() {
+    const key = 'mymodels';
+    const headerLabel = 'My models';
+    const labels = [
+      'Name', 'Machines, Cloud/region', 'Last accessed', 'Action'];
+    const tableHeader = this._generateTableHeader(labels, key);
+    const state = this.state.models;
+    const modelList = state && state.owned;
+    if (!modelList || modelList.length === 0) {
+      return [this._generateHeader(headerLabel, 0, true), tableHeader];
+    }
+    const rowData = modelList.reduce((models, model) => {
+      // Keep only the models that aren't currently in the destroy cycle.
+      if (!model.isAlive) {
+        return;
+      }
+      models.push([
+        model.name,
+        `${model.numMachines} ${model.provider.toUpperCase()}/${model.region.toUpperCase()}`, // eslint-disable-line max-len
+        model.lastConnection,
+        '-'
+      ]);
+      return models;
+    }, []);
+    const rows = this._generateRows(rowData, key);
```

Comment:
Right now there is just a header with no content - this will be updated and tested for once we have a better UI for that eventuality.